presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/classloader/ClassLoaderSafeConnectorPageSourceProvider.java

```diff
@@ -42,14 +42,6 @@ public ClassLoaderSafeConnectorPageSourceProvider(@ForClassLoaderSafe ConnectorP
         this.classLoader = requireNonNull(classLoader, "classLoader is null");
     }
 
-    @Override
-    public ConnectorPageSource createPageSource(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorSplit split, ConnectorTableHandle table, List<ColumnHandle> columns)
```

Comment:
i'd prefer to keep it:

- it safes mental overhear required to justify not having it
- someone could use this class in their code and now would lose TCCL wrapping

is there a benefit from removing it?

Reply:
- this method was never called in Presto

- copy&paste is also bad, makes even more confusion, especially that somebody can take this class as an example for their own implementations

- they will have to be removed at some time in future - why promote bad behavior

- the case when somebody calls deprecated method is rare, but indeed can be serious; maybe the pattern should be modified a bit:
```java
interface Xxxxx {
  @Deprecated
  default void xxxxx(...) {
    try {
      xxxxx(...);  // call non-deprecated version (if exists)
    } catch(StackOverflowException e) { 
       throw new UnsupportedOperationException("xxxxx not implemented");
    }

   void xxxxx(...) {
      xxxxx(...); // call deprecated version (if exists)
   }
}
```

I know this is not the most beautiful code in the world, but for @Deprecated stuff, to be removed in the future, I think it's pretty acceptable.

Reply:
I suggest we leave the method, with `throw Unsupported`  as the implementation.
This way we have the safety you want, and also the readability provisions I want.
And there is no need to relax the assertions in the test.

Reply:
there are many more places like this, with even more serious problems, my suggestion how to solve this once and for all is here: https://github.com/prestosql/presto/issues/4448

Reply:
Thanks for linking the two together.

I feel strongly that `ClassLoaderSafe*` should just dutifully implement all methods.